activerecord/lib/active_record/persistence.rb

```diff
@@ -1167,6 +1171,13 @@ def apply_scoping?(options)
         (self.class.default_scopes?(all_queries: true) || self.class.global_current_scope)
     end
 
+    def build_reload_scoping_constraint(current_scope)
+      default_scope_clause = self.class.default_scoped(all_queries: true).where_clause
+      default_scopes_to_unscope_clause = self.class.default_scoped.where_clause
+      reload_clause = (current_scope.where_clause - default_scopes_to_unscope_clause) + default_scope_clause
```

Comment:
We need to maintain all filtering in the current scope, remove all default scopes, and only add default scopes with `all_queries: true`.